Getting started - Linux (#141)
microsoft/vcpkg-docs: vcpkg/get_started/setup-vcpkg.md

> ---

Reviewer: It might be best to just pull this file out into a separate PR. We will want to make it applicable to all 3 operating systems using tabs for the user to swap between them. It will need a bit of work so it's easier to review it that way.

> title: Setup vckg on Linux

Reviewer: ```suggestion
title: Install vcpkg
```

> description: Tutorial guides the user through the process of setting up vcpkg on Linux.

Reviewer: ```suggestion
description: Tutorial for installing vcpkg on Windows, macOS, or Linux systems
```

author: JavierMatosD
ms.date: 08/31/2023
ms.author: javiermat
ms.prod: vcpkg
---

> # Setup vcpkg on Linux

Reviewer: ```suggestion
# Install vcpkg
```

## Description

> This tutorial provides a step-by-step guide on setting up vcpkg, a C++ package manager, on a Linux system. It covers everything from cloning the vcpkg repository and running the bootstrap script to integrating vcpkg into the bash shell for tab-completion features.

Reviewer: ```suggestion
This tutorial provides a step-by-step guide for installing vcpkg on a Windows, macOS, or Linux system. You should follow this tutorial if you want to set up vcpkg independently on a development machine or continuous integration system. If you are a Visual Studio or Visual Studio Code user, see the following tutorials for more specific instructions: 
* [Use vcpkg from Visual Studio IDE](vcpkg-VS.md)
* [Use vcpkg from Visual Studio Code](vcpkg-VS-Code.md)
```

## Clone the repository

> The first step is to clone the vcpkg repository. The repository contains scripts to acquire the vcpkg executable and a registry of open-source packages that is curated and maintained by the vcpkg team and it's community of contributors. To do this, run:

Reviewer: ```suggestion
The first step is to clone the vcpkg repository from GitHub. The repository contains scripts to acquire the vcpkg executable and a registry of curated open-source libraries maintained by the vcpkg community. To do this, run:
```

```bash
git clone https://github.com/microsoft/vcpkg.git
```

> The curated registry is a set of libraries that have been vetted and packaged to work smoothly with vcpkg. vcpkg uses "ports" to package open-source projects and make them available to install. Ports typically contain:

Reviewer: ```suggestion
The vcpkg open-source registry is a set of over 2,000 free libraries that have been tested together and optimized to work smoothly with vcpkg. While the vcpkg repo does not contain the source code for these libraries, it does contain build recipes and metadata about each library. These files will be included when vcpkg is installed to your system. 
```

> 1. `portfile.cmake`: A CMake script that contains instructions for downloading, building, and installing the library.
> 
> 2. `vcpkg.json`: A JSON file including package metadata: package version, dependencies, supported platforms and features, etc.
> 
> 3. Patches: Sometimes a port will contain patches to make the library work with vcpkg or to provide bug fixes in the library itself.

Reviewer: ```suggestion
1. `portfile.cmake`: A CMake script that contains instructions for downloading, building, and installing the library.

2. `vcpkg.json`: A JSON file including package metadata: package version, dependencies, supported platforms and features, etc.

3. Patches: Sometimes a port will contain patches to make the library work with vcpkg or to provide bug fixes in the library itself.
```
I think this can be removed, it's too much information for the installation tutorial.

> 

Reviewer: I would add a line about how to check if the clone was successful (what the user should see).

> Once you have the repository, navigate to the `vcpkg` directory and execute the bootstrap script:

Reviewer: ```suggestion
Now that you have cloned the vcpkg repository, navigate to the `vcpkg` directory and execute the bootstrap script:
```

> ## Bootstrap and integrate

Reviewer: ```suggestion
## Bootstrap vcpkg and integrate your preferred environment
```

```bash
cd vcpkg && ./bootstrap-vcpkg.sh
```

The shell script performs prerequisite checks, downloads the vcpkg executable, and manages telemetry settings.

You'll want to integrate vcpkg into your bash shell to enjoy tab-completion features. To do this, run:

```bash
./vcpkg integrate bash
```

The `vcpkg integrate bash` command sets up bash shell integration for vcpkg, which adds vcpkg tab-completion support to the current user's `.bashrc`.

> That's it! vcpkg is setup and ready to use.

Reviewer: ```suggestion
That's it! vcpkg is set up and ready to use.
```
